Alignment/HIPAlignmentAlgorithm/src/HIPAlignmentAlgorithm.cc

```diff
@@ -743,769 +730,787 @@ void HIPAlignmentAlgorithm::run(const edm::EventSetup& setup, const EventInfo &e
-	
-  // errors of parameters
-
- if (setDet==0){
-  int ierr;
-  AlgebraicSymMatrix jtvjinv = jtvj.inverse(ierr);
-
-  if (ierr !=0) {
-    edm::LogError("Alignment") << "Matrix inversion failed!";
-    return false;
-  }
-  params = - (jtvjinv * jtve);
-  cov = jtvjinv;
-
-//  edm::LogWarning("Alignment") << "[HIPAlignmentAlgorithm::CalcParameters]: parameters before RelErrCut= " << params;
-//  edm::LogWarning("Alignment") << "[HIPAlignmentAlgorithm::CalcParameters]: cov= " << cov;
-  AlgebraicVector relerr(npar);
-  for (int i=0;i<npar;i++) {
-    if (fabs(cov[i][i])>0) paramerr[i] = sqrt(fabs(cov[i][i]));
-    else paramerr[i] = params[i];
-    if (params[i]!=0) relerr[i] = fabs(paramerr[i]/params[i]);
-    else relerr[i]=0;
-    if (relerr[i] >= theMaxRelParameterError) { 
-//   edm::LogWarning("Alignment") << "RelError="<<relerr[i]<<"too large!" ;
-    params[i] = 0; 
-    paramerr[i]=0; 
-     }
+      edm::LogWarning("Alignment") << "@SUB=HIPAlignmentAlgorithm::calcParameters" << "Parameters = " << params;
     }
   }
 
-  if (setDet!=0)
-  edm::LogWarning("Alignment") << "[HIPAlignmentAlgorithm::CalcParameters]: parameters = " << params;
-
   uservar->alipar=params;
   uservar->alierr=paramerr;
 
-  AlignmentParameters* parnew = par->cloneFromSelected(params,cov);
+  AlignmentParameters* parnew = par->cloneFromSelected(params, cov);
   ali->setAlignmentParameters(parnew);
   parnew->setValid(true);
 
+  edm::LogInfo("Alignment") << "@SUB=HIPAlignmentAlgorithm::calcParameters" << "End: Processing detector " << ali->id();
+
   return true;
 }
 
 //-----------------------------------------------------------------------------
+void HIPAlignmentAlgorithm::collector(void){
+  edm::LogInfo("Alignment") << "@SUB=HIPAlignmentAlgorithm::collector" << "Called for iteration " << theIteration;
 
-void HIPAlignmentAlgorithm::collector(void)
-{
-  edm::LogWarning("Alignment") << "[HIPAlignmentAlgorithm::collector] called for iteration "
-			       << theIteration << std::endl;
-	
+  std::vector<std::string> monitorFileList;
   HIPUserVariablesIORoot HIPIO;
 
-	
-  for (int ijob=1;ijob<=theCollectorNJobs;ijob++) {
-		
-    edm::LogWarning("Alignment") << "reading uservar for job " << ijob;
-    
+  typedef int pawt_t;
+  std::unordered_map<Alignable*, std::unordered_map<int, pawt_t> > ali_datatypecountpair_map;
+  if (rewgtPerAli){
+    edm::LogInfo("Alignment") << "@SUB=HIPAlignmentAlgorithm::collector" << "Per-alignable reweighting is turned on. Iterating over the parallel jobs to sum number of hits per alignable for each data type.";
+    // Counting step for per-alignable reweighting
+    for (int ijob=1; ijob<=theCollectorNJobs; ijob++){
+      edm::LogInfo("Alignment") << "@SUB=HIPAlignmentAlgorithm::collector" << "Pre-reading uservar for job " << ijob;
+
+      std::stringstream ss;
+      std::string str;
+      ss << ijob;
+      ss >> str;
+      std::string uvfile = theCollectorPath+"/job"+str+"/"+suvarfilecore;
```

Comment:
Possible, should I make a separate PR (or add it to the one I proposed) for this instead of making changes in this PR?